test.py

```diff
@@ -20,6 +21,8 @@ def setUp(self):
     def tearDown(self):
         if 'cuda' in str(self):
             memory = torch.cuda.memory_allocated()
+            gc.collect()
```

Comment:
and try removing empty_cache, and moving gc.collect() above one line.  this is to make sure any collectible tensors are collected _before_ we take the measurement not after.